google/cloud/storage/client.cc

```diff
@@ -481,6 +481,10 @@ std::string CreateRandomPrefixName(std::string const& prefix) {
                                                   "abcdefghijklmnopqrstuvwxyz");
 }
 
+std::string Client::ExternalUrl() {
```

Comment:
Are you planning to use this function in a second call site?  Otherwise it does not seem to carry its weight.

```suggestion
std::string Client::ExternalUrl() const {
```